.azure-pipelines/pipelines.yml

```diff
@@ -23,6 +23,8 @@ variables:
 - name: isStableBranch
   # A release branch can be either `main` or a `release/v1.x` stable branch
   value: $[or(eq(variables['isMain'], 'true'), eq(variables['isReleaseBranch'], 'true'))]
+- name: nonMobileFilesChanged
+  value: git diff --name-only origin/main | grep -qe common/
 
 stages:
```

Comment:
```suggestion
stages:
- stage: change_detection
  jobs:
  - job: format
    dependsOn: []
    pool:
      vmImage: "ubuntu-20.04"
    steps:
    - bash: |
        git diff --name-only origin/main | grep -qe common/ \
          || echo "##vso[task.setvariable variable=nonMobileFilesChanged]False"
      displayName: "Detect repo changes"
      workingDirectory: $(Build.SourcesDirectory)

```

Reply:
`git diff --name-only origin/main | grep -qe common/` this doesnt seem right - but i think this is ~what we need

i believe this will effect subsequent stages but not sure i can test further

incidentally the pattern is pretty similar with github actions iirc - although not sure if we can do that once across all jobs

would be good to turn off mobile linting etc when its an envoy only build

Reply:
also not 100% about bools in azp yaml

Reply:
we might need/want to force other stages to depend on this one